include/libswiftnav/signal.h

```diff
@@ -15,68 +16,85 @@
 
 #include <libswiftnav/common.h>
 
+/** \addtogroup signal
+ * \{ */
+
+/* Number of satellites in each constellation. */
 #define NUM_SATS_GPS 32
 #define NUM_SATS_SBAS 19
 #define NUM_SATS (NUM_SATS_GPS + NUM_SATS_SBAS)
 
+/* Number of codes in each constellation. */
+#define NUM_CODES_GPS 2
```

Comment:
I think they could be useful if we need code-specific lookups for a single constellation. I did add a test to make sure the NUM_CODES_\<constellation\> defines are consistent with what code_to_constellation() returns.

Hopefully NUM_SIGNALS_\<code\> will always equal NUMS_SATS_\<constellation\>, but I do like the abstraction. If we ever add another field or something I could see it changing.